aip/general/0140.md

```diff
@@ -88,6 +88,31 @@ For uniformity, field names that contain both a noun and an adjective
 - `collected_items` (**not** `items_collected`)
 - `imported_objects` (**not** `objects_imported`)
 
+### Verbs
+
+Field names **must not** be expressed to reflect an intent or action.
+That is, they **must not** be expressed as verbs. Rather, the field name
+should express the condition of that aspect of the resource as defined
+by the value of the field in question.
+
+When performing a mutation, the field value is set to the desired
+*final* value of that property. When performing a read, the field value
+is returned as the *current* value of that property. As such, the name
+should always be *a noun or an adjective* as it describes what is so,
+not what to do.
+
+- `collected_items` (**not** `collect_items`)
+- `imported_objects` (**not** `import_objects`)
+
+Alternatively, method names, whether standard or custom, express intent
+to change the condition of resources and **should** be named as verbs,
+accordingly.
+
+**Note:** Fields named as verbs may suggest that there should be a
+complementary, corresponding field that is a noun or adjective, one to
+set a value and the other to get the value, respectively. This is highly
```

Comment:
I actually think this is the pattern we plan on recommending. @bgrant0607 for thoughts.

But I think you're referring to the idea of the input field being a verb and the output field being a noun or an adjective? which I agree isn't good.

I feel like you don't even need this paragraph. Your above explanation for how resources represent end state and therefore shouldn't include nouns is convincing as-is.

Reply:
I'm not sure I understand the question.

My canonical example is the Kubernetes Deployment spec "paused" field. "paused" not "pause". That matches this proposal, I believe.

The "double simplex" approach we've been discussing to address static defaults, semi-static defaults, configured defaults, dynamic defaults, semantic strings, associative lists, value normalizations (e.g., lower casing or upper casing), value additions, and other service-side transformations is a different issue, I believe. 

Example: `foo` and output-only `effective_foo`.

Unless this is a reference to the desired_state proposal or similar case where the service may want to return status information regarding whether the desired state has been achieved.

Reply:
@bgrant0607 I think we were asking more about the "double simplex" approach and I think you answered it. We were discussing the paragraph (removed) about that it is a poor practice to have two fields--one with a verb name and one with a noun name--because it is not IaC friendly. However, we now realized that this is a good pattern for the cases that you mentioned. But, as I said, this paragraph was removed and we appreciate your input here.